src/main/scala/com/ubirch/responder/ResponderMicroservice.scala

```diff
@@ -41,7 +41,7 @@ class ResponderMicroservice(runtime: NioMicroservice[Either[String, MessageEnvel
   def handleNormal(record: ConsumerRecord[String, MessageEnvelope]): ProducerRecord[String, MessageEnvelope] = {
 
     val tryResponseUPP = for {
-      requestId <- Try(record.key())
+      requestId <- Try(record.requestIdHeader().get)
```

Comment:
Is it because it's a Try() that you didn't use the OrNull like elsewhere?

Reply:
Yes, as we are using a for-comprehension and at this point we want to use the request id as the body of the response, I'd rather have it fail at that point than later in case it hasn't been set.